gate-core/src/main/groovy/com/netflix/spinnaker/gate/services/PermissionService.groovy

```diff
@@ -128,6 +142,49 @@ class PermissionService {
     }.execute() as Set<Role>
   }
 
+  private List<UserPermission.View> lookupServiceAccounts(String userId) {
+    try {
+      return extendedFiatService.getUserServiceAccounts(userId)
+    } catch (RetrofitError re) {
+      boolean notFound = re.kind == RetrofitError.Kind.HTTP && re.getResponse().getStatus() == HttpStatus.NOT_FOUND.value()
+      if (notFound) {
+        return []
+      }
+      boolean shouldRetry = re.kind != RetrofitError.Kind.HTTP || !HttpStatus.valueOf(re.getResponse().getStatus()).is4xxClientError()
```

Comment:
Are there not some types of `RetrofitError.Kind.HTTP` we want to retry?  I think `RetrofitError.Kind.HTTP` is any non-200 error, I think we'd want to retry 500s?

Reply:
Oh sorry, didn't scroll my browser window over.  The second conditional will return true on 500s.

Reply:
what I intended here was retry if it is a non HTTP error or if it is a non 4xx (I don't think we have anything currently rate-limiting where we would want to retry on a 429 or something)

I *think* I wrote it that way but I should probably add some test cases around this

Reply:
Yea I think it looks correct, just took me a moment to realize what you were going for.

Reply:
I added some tests in here and reworked the check a bit (it is possible to also get a RetrofitError ok Kind Conversion with an http status code, so I based it off the existance of a Response rather than off the Kind)